netbox/dcim/forms.py

```diff
@@ -2725,6 +2725,23 @@ class Meta:
             'termination_b_site', 'termination_b_rack', 'termination_b_device', 'termination_b_id', 'type', 'status',
             'label', 'color', 'length', 'length_unit',
         ]
+        widgets = {
+            'type': StaticSelect2(),
+            'status': StaticSelect2(),
+            'length_unit': StaticSelect2(),
+        }
+
+    def __init__(self, *args, **kwargs):
+
+        super().__init__(*args, **kwargs)
+
+        # Initialize site and rack fields to match that of the existing temination
+        if self.instance and self.instance.termination_a:
+            termination_a_parent = self.instance.termination_a.parent
+            if hasattr(termination_a_parent, 'site'):
+                self.initial['termination_b_site'] = termination_a_parent.site
+            if hasattr(termination_a_parent, 'rack'):
+                self.initial['termination_b_rack'] = termination_a_parent.rack
```

Comment:
The rack isn't getting pre-selected in the form widget, even though the initial data is being passed correctly. I'm sure this has something to do with the APISelect widget's behavior. Not sure it's worth tracking down but that's up to you.

Reply:
This is happening because of this and is affecting any chained field. Currently trying to figure out what's the purpose of that.
https://github.com/netbox-community/netbox/blob/66daeda85f5cf1e5b2b84dbb1027653af5af9425/netbox/utilities/forms.py#L724

Reply:
Correct me if I'm wrong, but I think `ChainedFieldsMixin` is no longer needed as it's been replaced with the `filter_for` functionality of APISelect? If that's the case, should we cleanup the chaining? If so, may I 😄 ?

Reply:
The rack issue will be resolved as part of #3912